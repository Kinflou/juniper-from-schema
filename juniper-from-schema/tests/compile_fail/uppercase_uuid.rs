#![allow(dead_code, unused_variables, unused_must_use, unused_imports)]
include!("../compile_pass/setup.rs");

use uuid::Uuid;

juniper_from_schema::graphql_schema! {
    schema {
        query: Query
    }

    type Query {
        uuid: UUID! @juniper(ownership: "owned")
    }

    scalar UUID
}

pub struct Query;

impl QueryFields for Query {
    fn field_uuid(&self, _: &Executor<Context>) -> FieldResult<Uuid> {
        Ok(Uuid::new_v4())
    }
}
