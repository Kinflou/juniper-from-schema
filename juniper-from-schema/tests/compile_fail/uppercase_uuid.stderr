error: proc macro panicked
  --> $DIR/uppercase_uuid.rs:6:1
   |
6  | / juniper_from_schema::graphql_schema! {
7  | |     schema {
8  | |         query: Query
9  | |     }
...  |
15 | |     scalar UUID
16 | | }
   | |_^
   |
   = help: message:

           error: The UUID must be named `Uuid`
            --> schema:2:50
             |
           2 |    { uuid : UUID ! @ juniper(ownership : "owned") } scalar UUID
             |                                                     ^

           This is to be consistent with the naming the "uuid" crate

           aborting due to previous error

error[E0405]: cannot find trait `QueryFields` in this scope
  --> $DIR/uppercase_uuid.rs:20:6
   |
20 | impl QueryFields for Query {
   |      ^^^^^^^^^^^ not found in this scope
